apps/web/src/components/WidgetIconResolver/WidgetIconResolver.jsx

```diff
@@ -2,12 +2,12 @@ import React from 'react'
 import Avatar from 'components/Avatar'
 import { Grid3x3 } from 'lucide-react'
 import Icon from 'components/Icon'
-import { widgetTypeResolver } from 'util/contextWidgets'
+import { widgetTypeInferrer } from '@hylo/shared/src/ContextWidgetPresenter'
 import { ViewHelpers } from '@hylo/shared'
 
 export function WidgetIconResolver ({ widget, style, className }) {
   if (!widget) return null
-  const type = widgetTypeResolver({ widget })
+  const type = widgetTypeInferrer({ widget })
```

Comment:
The idea of the presenter though is that you'd have this method on the ContextWidgetPresenter and you'd transform a widget as soon as you see it like

`const widget = ContextWidgetPresenter(providedFoundQueriedWhatverWidget)`

That is the scoping I'm expecting. Is there some reason to not do that? It does have a lot of advantages in my view which I'd happily talk through with you if you see something I don't.

Reply:
As you have a collection I would do the *Presenter transform on the whole collection where you can so you don't have to call the Presenter throughout the code. e.g.: 

`const widgets = mywidgetcollection.map(widget => ContextWidgetPresenter(widget))`.

You could also just make a usePresentedWidgets hook or something. The idea with Presenters is that they happen pretty high up in the component where possible, as the directly they may be going is that they are applied in URQL before the component like what happens with Reducx ORM  (just not ready to go there yet). Keeping tight on our use of them now by always transforming the whole object, and doing it early in the component will leave us in a place to switch over to that later fairly easily.

Reply:
I was hanging on including inferredType in the presenter because I hadn't checked all the wider impacts. Done that now.

Reply:
But that will be put in the presenter, right? So you will end up with a transformed widget that will have widget.inferredType? 

Also, on the presenter should resolve iconName. Even if you still need this special WidgetIcon component for other reasons, that logic is going to be used in Mobile as well.

Reply:
just resolving iconName isn't enough.

I'm using a separate 'icon' resolution component, because it actually might return several categories of components; Avatars, Lucide icons or the prior icon